### PR TITLE
Restart Tendermint node if no begin block requests are received within a timeout

### PR DESCRIPTION
**Pull request by Adamantios**
## Proposed changes

Sets a timeout for not receiving any block arrivals. Creates a policy that if we don't receive a block before that timeout, then we will restart the Tendermint node.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [x] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

## Further comments

[Trello ticket](https://trello.com/c/HBnO417E).

### CODE REVIEW

**Review by dvilelaf — DISMISSED** [review]
LGTM

**Review by angrybayblade — DISMISSED** [review]
LGTM

**Review by DavidMinarsch — COMMENTED** [review]
Please look into my comment.

File: packages/valory/skills/abstract_round_abci/behaviour_utils.py [hunk 1]
```diff
@@ -593,6 +600,17 @@ def send_a2a_transaction(
...
+        if self.tm_communication_unhealthy:
+            self.context.logger.warning(
+                "The local deadline for the next `begin_block` request from the Tendermint node has expired! "
+                "Trying to reset local Tendermint node as there could be something wrong with the communication."
+            )
+            # if we end up looping here forever,
+            # then there is probably something serious going on with the communication
+            reset_successfully = yield from self.reset_tendermint_with_wait()
+            if not reset_successfully:
+                return
```
[the remark this answers is not in the document]
Comment (Adamantios) [reply packages/valory/skills/abstract_round_abci/behaviour_utils.py]: Refactored in 4c7176a.

**Review by DavidMinarsch — APPROVED** [review]
lgtm

**Pull request merged**